src/lib/spinel/spinel_interface.hpp

```diff
@@ -35,12 +35,10 @@
 #ifndef POSIX_APP_SPINEL_INTERFACE_HPP_
 #define POSIX_APP_SPINEL_INTERFACE_HPP_
 
-#include "openthread-posix-config.h"
-
 #include "lib/hdlc/hdlc.hpp"
```

Comment:
I would expect this spinel interface header has no dependencies.

Reply:
Unfortunately it has to be there.
We are sharing a buffer between Spinelnterface and the radio and this header is for the type of that buffer.
It's not a good design to share private members but I'm afraid it will be too radical a change to fix it in this PR.

Reply:
Then we need to change the CMake file to be able to find the `hdlc.hpp` header, when current header is  included out of openthread codebase.